### DISCUSSION TITLE
Getting Content From Editor

### DISCUSSION THREAD

**Discussion by raknay** [umaranis/svelte-lexical#45] [Q&A] [answered]
Hi, I'm quite new to Svelte and frontend framework in general. I was trying to create a basic CMS site with `svelte-lexical`. How to get contents from the editor(as HTML or any other suitable format) so that I can save it to database.

**Comment by umaranis** [comment, marked as answer]
Hi @raknay ,

If you are using `RichTextComposer`, you need to get a reference to it using `bind:this={composer}`, then call `composer.getEditor().getEditorState()`. 
This is going to return the state in JSON format.

Here is a similar example that prints editor contents to the console log every time the content changes.
https://github.com/umaranis/svelte-lexical/blob/f2b1f63cac29bb16e93186fe128fae2528eab723/demos/playground/src/App.svelte

**Reply by enesbala5** [reply]
Please explain how to do this in the Sveltekit version

**Reply by umaranis** [reply]
@enesbala5 please upgrade to version 0.3.3 and you will be able to use the same logic for Sveltekit also.

**Reply by milansimek** [reply]
@umaranis For me it also wasn't clear at all how to get the value. I almost decided to use another library instead (but this one is just too sweet :))

Maybe it would be a good idea to:
- Add this info to the docs
- Move docs to Wiki so they can be found more easily
- Add `export let value;` inside the component (which is the standard way to retrieve data from Svelte components)
- Update the demos accordingly

What do you think? I'd be happy to make some changes and create a pull request if you agree.

**Reply by umaranis** [reply]
Hi @milansimek ,

It would be great if you could make the suggested changes and submit the pull request. I am looking forward to it.

I think we really need a getting started doc page to start with.
